src/Model/DocumentModel.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef _DOCUMENT_MODEL_HPP_
+#define _DOCUMENT_MODEL_HPP_
+
+#include "Automerge.h"
+#include "JsonSchemaValidator.hpp"
+
+class DocumentModel
+{
+public:
+  DocumentModel(const nlohmann::json& schema, const nlohmann::json& document);
+  ~DocumentModel();
+
+  void addAt(const json::json_pointer& path, const json& value);
+  void replaceAt(const json::json_pointer& path, const json& value);
+  void deleteAt(const json::json_pointer& path);
+
+  json documentJson();
```

Comment:
1. should add "const" suffix to declaration if it is a copy
2. however, copy of the whole document is less efficient

Consider redesign this API for retrieving data.